packages/test/test-end-to-end-tests/src/test/connectionMode.spec.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { MockLogger } from "@fluidframework/telemetry-utils";
+import { delay } from "@fluidframework/common-utils";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+}
+
+describeNoCompat("Connection Mode", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+    );
+
+    let mainContainer: IContainer;
+
+    const createContainer = async (logger): Promise<IContainer> => provider.createContainer(runtimeFactory, { logger });
+
+    beforeEach(async () => {
```

Comment:
i don't think a before each is necessary. just put it all in the test to keep the test suite simple